jcl/src/openj9.diagnostics/share/classes/openj9/tools/attach/diagnostics/Jps.java

```diff
@@ -0,0 +1,131 @@
+	private static boolean printApplicationArguments;
+	private static boolean printJvmArguments;
+	private static boolean noPackageName;
+	private static boolean vmidOnly;
+	/**
+	 * Print a list of Java processes and information about them.
+	 * @param args Arguments to the application
+	 */
+	public static void main(String[] args) {
+		int rc = 0;
+		parseArguments(args);
+		AttachProvider theProvider = AttachProvider.providers().get(0);
+		if (null == theProvider) {
+			System.err.println("no attach providers available"); //$NON-NLS-1$
+			rc = 1;
+		} else {
+			ArrayList<String> outputBuffer = new ArrayList<>();
+			List<VirtualMachineDescriptor> vmds = theProvider.listVirtualMachines();
+			for (VirtualMachineDescriptor vmd: vmds) {
+				outputBuffer.clear();
+				outputBuffer.add(vmd.id());
+				if (!vmidOnly) {
+					try {
+						VirtualMachine theVm = theProvider.attachVirtualMachine(vmd);
+						try {
+						Properties vmProperties = theVm.getSystemProperties();
+						String theCommand = vmProperties.getProperty(SUN_JAVA_COMMAND, ""); //$NON-NLS-1$
+						String parts[] = theCommand.split("\\s+", 2); /* split into at most 2 parts: command and argument string */  //$NON-NLS-1$
+						if (noPackageName && (parts.length > 0)) {
+							String commandName = parts[0];
+							int finalDot = commandName.lastIndexOf('.');
+							parts[0] = commandName.substring(finalDot+1); /* if the dot is missing, we get the whole string */
+						}
+						if (printApplicationArguments) {
+							outputBuffer.addAll(Arrays.asList(parts));
+						} else {
+							if (parts.length > 0) { /* some Java processes do not use the Java launcher */
+								outputBuffer.add(parts[0]);
+							}
+						}
+						if (printJvmArguments) {
+							String jvmArguments = vmProperties.getProperty(SUN_JVM_ARGS);
+							if ((null != jvmArguments) && !jvmArguments.isEmpty()) {
+								outputBuffer.add(" "); //$NON-NLS-1$
+								outputBuffer.add(jvmArguments);
+							}
+						}
+						} finally {
+							theVm.detach();
+						}
+					} catch (AttachNotSupportedException | IOException e) {
+						outputBuffer.add("<no information available>"); //$NON-NLS-1$
+					}
+				}
+				System.out.println(String.join(" ", outputBuffer.toArray(new String[outputBuffer.size()]))); //$NON-NLS-1$
+			}
+		}
+		System.exit(rc);
+	}
+
+	@SuppressWarnings("nls")
+	private static void parseArguments(String[] args) {
+		printApplicationArguments = false;
+		printJvmArguments = false;
+		noPackageName = true;
+		vmidOnly = false;
+		final String HELPTEXT = "jps: Print a list of Java processes and information about them\n"
+				+ "-J: supply arguments to the Java VM running jps\n"
+				+ "-l: print the application package name\n"
+				+ "-q: print only the virtual machine identifiers\n"
+				+ "-m: print the application arguments\n"
+				+ "-v: print the Java VM arguments, including those produced automatically";
+		for (String a: args) {
+			switch (a) {
+			case "-J": break;
```

Comment:
I don't think this is required, as any -J argument should be stripped before this code runs.

Reply:
Fixed.